en/lessons/jupyter-notebooks.md

```diff
@@ -0,0 +1,391 @@
+
+Paste the following into the first cell. If it doesn't appear with the first line in a large font (as a header), make sure you've selected "Markdown" from the dropdown at the top.
+
+```
+# Fanfic date conversion
+Converting published & updated dates for Italian fanfic into days of the week.
+```
+
+{% include figure.html filename="jupyter-editing.png" caption="Editing Markdown cell in a Jupyter notebook" %}
+
+When you're editing a cell, you can use Ctrl + Z (Win) or Command + Z (Mac) to undo changes that you've made. Each cell retains its own edit history; even if you move to a different cell and make edits there, you can subsequently click back on the first cell and undo your earlier changes there, without losing the changes to the second cell.
+
+To leave editing mode and "run" this cell (for a Markdown cell, this doesn't *do* anything, just moves you further down the notebook), you can click <i class="fa-step-forward fa"></i> in the toolbar or press Ctrl+Enter (Ctrl+Return on Mac). If you want to resume editing it later, you can either double-click it, or select the cell (which will show a vertical blue line on the left once it's selected) by clicking it once, and then hitting the Enter (Win) or Return (Mac) key. To leave editing mode, you can click <i class="fa-step-forward fa"></i> in the toolbar or press Ctrl+Enter (Ctrl+Return on Mac). If you want to run your current cell and add a new cell (by default, a code cell) immediately below it, you can press Alt+Enter (Option+Enter on Mac).
+
+Next, you need to figure out how to do the conversion. Searching for relevant terms may lead you to [this StackOverflow discussion](https://stackoverflow.com/questions/2265357/parse-date-string-and-change-format), and the first answer involves using the *datetime* Python module. As a first step, you need to import *datetime*, using a code cell. You also know that your input file is a CSV, so you should import the *csv* module as well.
+
+To add a new cell, click the plus button <i class="fa fa-plus"></i> in the toolbar (or use the *esc + b* keyboard shortcut). This will create a new code cell below the cell that's currently selected. Create a new code cell, and paste in the following code to import a Python module:
+
+```
```

Comment:
In all the code blocks that contain python, we can enable code highlighting by appending `py` after the three backticks

```
``` py
```

(make sure to include the space between the 3rd backtick and the `py`)

This will help make the code chunks, especially the very long ones, much more readable - and it will better match the formatting that students will see in the Jupyter interface, too!

Reply:
Oh, very cool! I just added it to the python codeblocks.